lib/modules/manager/terragrunt/lockfile.ts

```diff
@@ -0,0 +1,12 @@
+import { updateArtifacts as updateTerraformArtifacts } from '../terraform/lockfile/index';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+export async function updateArtifacts(
+  artifact: UpdateArtifact
+): Promise<UpdateArtifactsResult[] | null> {
+  if (artifact.config.updateType !== 'lockFileMaintenance') {
```

Comment:
why only lockfile maintenance?

Reply:
I was following the advice of @secustor in [this comment](https://github.com/renovatebot/renovate/issues/13393#issuecomment-1007957758).

Are there other update types that this wrapper should accept?

Reply:
don't you also need to update the lockfile when the provider version changes?

with current implementation the lockfile would only be updated on lockfile maintenance.

Reply:
@secustor do you know more on terragrunt tooling?

Reply:
I have a rough idea regarding it, but no details. I will look into it.

Reply:
I think we want renovate to update the lockfile any time it would do so if the directory were a normal terraform configuration. As far as I can tell, that means there is no restriction on `updateType`, and this code becomes a lot simpler.

Reply:
terraform manager only updates lockfiles if provider version changes or lockfile maintenance

Reply:
I've updated this branch to remove the wrapper function, and to instead import the whole terraform `updateArtifacts`. I'll update the PR description shortly, but I've run the tests and code coverage is maintained at pre-branch levels.

Reply:
Lockfile updates on dependency updates are currently not possible without some bigger additions to the code.    

Artifact update work currently with a relation from the package file to lock files.     
In case of Terraform this is from a tf file with a reference to  a `.terraform.lock.hcl` in the same folder.    
For Terragrunt these files are `terragrunt.hcl` and `.terraform.lock.hcl`, which have to be again in the same folder.

Therefore there has to be an update in `terragrunt.hcl` to trigger an update in `.terraform.lock.hcl`. `lockfileMaintenance` is an exception for this as here only the relation has to be detected.    
If this has happened, the constraints are used to find a fitting version. 

Therefore you have to ensure that only updateType with `lockfileMaintenance` as value leads to a execution of the `updateArtifacts` function of the `terraform` manager.

Reply:
I think I understand this so far. Let me ask something new to test my understanding.

Say I have a `terragrunt.hcl` with the following contents:

```
terraform {
  source = "git@github.com:user/repo.git//path/to/module?ref=1.0.0"
}
```

This module at version 1.0.0 requires the `aws` provider version `3.x`.

Now let's say that module releases a new version `2.0.0`, which updates its required provider version to `4.x`.

When Renovate runs to upgrade my Terragrunt configuration to the latest version of the module, that upgrade must include a change to the lockfile artifact.

In this scenario, did Renovate run with an updateType of `lockfileMaintenance`?

Reply:
> In this scenario, did Renovate run with an updateType of lockfileMaintenance?

No, in this specific case it will run with `major` as `updateType`, because Renovate updates the module from `1.0.0` to `2.0.0`. It has no knowledge what is in the module, it only looks at the version it has currently updated. 

> [perhaps this question is evidence that I should be building a fixture and integration test...]

A good indicator is that whenever you change the behaviour it should be reflected in some tests.

Reply:
> It has no knowledge what is in the module, it only looks at the version it has currently updated.

So in this case, a `major` update would require further (manual) intervention before it could be merged, which seems like reasonable behavior.

If we change the scenario slightly my point might be better Illustrated. Say that the module is upgrading from version `1.0.0` to `1.1.0`, and its underlying provider requirements are changing from `3.76.0` to `3.76.1`. From Renovate's perspective, this is a `minor` updateType, and only affects the module version in `terragrunt.hcl`, it does not make changes to the lockfile. This scenario _also_ requires manual intervention before it can be reliably deployed, which I think is not reasonable behavior. I would expect anyone doing a "routine upgrade" to a module (such as a minor version upgrade) to update any lockfile artifacts as necessary. Does this go against Renovate policy or ideology?

Reply:
I've performed a new manual test. I ran Renovate as a `minor` update to both a terraform configuration and a terragrunt configuration. I [first did this](https://github.com/spilliams/terragrunt-renovate-test/pull/14) from the `main` branch of my fork of the repo, and [then from this feature branch](https://github.com/spilliams/terragrunt-renovate-test/pull/15). The behavior was the same in both cases: Renovate changed the version number of the module, but did nothing to the associated lockfile.

At this point I don't understand why a wrapper function with a condition for `config.updateType === 'lockFileMaintenance'` is necessary.  I would expect the Terragrunt manager to behave very much like the Terraform manager in all cases, and indeed my testing shows that it does behave the same.